regression/test.pl

```diff
@@ -117,10 +113,27 @@ ($$$$$)
           $included--;
         } else {
           my $r;
-          $result =~ s/\\/\\\\/g;
-          $result =~ s/([^\\])\$/$1\\r\\\\?\$/;
-          system("bash", "-c", "grep $grep_options \$'$result' '$name/$output' >/dev/null");
-          $r = ($included ? $? != 0 : $? == 0);
+
+          # Multi line therefore we run each check across the whole output
+          if($grep_options eq "activate-multi-line-match") {
+            my $dir = getcwd();
+            my $abs_path = "$dir/$name/$output";
+            open( my $fh => $abs_path) || die "Cannot open '$name/$output': $!";
+
+            local $/ = undef;
+            binmode $fh;
+            my $whole_file = <$fh>;
+            my $is_match = $whole_file =~ /$result/;
+            $r = ($included ? !$is_match : $is_match);
+            close($fh);
+          }
+          else
+          {
+            $result =~ s/\\/\\\\/g;
+            $result =~ s/([^\\])\$/$1\\r\\\\?\$/;
+            system("bash", "-c", "grep \$'$result' '$name/$output' >/dev/null");
```

Comment:
Full list:
```
Running struct6/test.desc  [FAILED]
Running struct7/test.desc  [FAILED]
Running BV_Arithmetic3/test.desc  [FAILED]
Running Fixedbv4/test.desc  [FAILED]
Running Malloc23/test.desc  [FAILED]
Running Multi_Dimensional_Array6/test.desc  [FAILED]
Running Overflow_Addition1/test.desc  [FAILED]
Running Pointer_byte_extract2/test.desc  [FAILED]
Running Pointer_byte_extract5/test.desc  [FAILED]
Running Quantifiers-assertion/test.desc  [FAILED]
Running Quantifiers-assignment/test.desc  [FAILED]
Running Quantifiers-copy/test.desc  [FAILED]
Running Quantifiers-if/test.desc  [FAILED]
Running Quantifiers-initialisation/test.desc  [FAILED]
Running Quantifiers-initialisation2/test.desc  [FAILED]
Running Quantifiers-invalid-var-range/test.desc  [FAILED]
Running Quantifiers-not/test.desc  [FAILED]
Running Quantifiers-not-exists/test.desc  [FAILED]
Running Quantifiers-two-dimension-array/test.desc  [FAILED]
Running Quantifiers-type/test.desc  [FAILED]
Running null3/test.desc  [FAILED]
Running pipe1/test.desc  [FAILED]
Running Address_of_Method2/test.desc  [FAILED]
Running Address_of_Method3/test.desc  [FAILED]
Running Constant2/test.desc  [FAILED]
Running Constant3/test.desc  [FAILED]
Running Pointer_Conversion1/test.desc  [FAILED]
Running Resolver12/test.desc  [FAILED]
Running Resolver4/test.desc  [FAILED]
Running union3/test.desc  [FAILED]
Running union4/test.desc  [FAILED]
Running union5/test.desc  [FAILED]
Running NullPointer1/test.desc  [FAILED]
Running NullPointer2/test.desc  [FAILED]
Running NullPointer3/test.desc  [FAILED]
Running NullPointer4/test.desc  [FAILED]
Running enum1/test.desc  [FAILED]
Running jar-file1/test.desc  [FAILED]
Running jar-file2/test.desc  [FAILED]
Running data-flow1/test.desc  [FAILED]
```

It appears to be caused (from the few I've checked) by brackets now needing to be escaped to check for brackets, and not escaped to check them as regex brackets. I guess can do some find and replaces:

`[^\\]\[` -> `\\\[`
`\\\[` -> `\[`

And same with `]`, `)`, `)` Will try this and report back.